server/stt_metrics_test.go

```diff
@@ -0,0 +1,70 @@
+package server
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	managementClient "github.com/percona/pmm/api/managementpb/json/client"
+	serverClient "github.com/percona/pmm/api/serverpb/json/client"
+	"github.com/percona/pmm/api/serverpb/json/client/server"
+	"github.com/prometheus/client_golang/api"
+	promapi "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	pmmapitests "github.com/Percona-Lab/pmm-api-tests"
+)
+
+const promAddress = "http://admin:admin@127.0.0.1:80/prometheus"
+
+func TestSTTMetrics(t *testing.T) {
+	client := serverClient.Default.Server
+
+	t.Run("StartSTTChecksAndRecordMetrics", func(t *testing.T) {
+		defer restoreSettingsDefaults(t)
+		// Enabled STT
+		res, err := client.ChangeSettings(&server.ChangeSettingsParams{
+			Body: server.ChangeSettingsBody{
+				EnableStt:       true,
+				EnableTelemetry: true,
+			},
+			Context: pmmapitests.Context,
+		})
+		require.NoError(t, err)
+		assert.True(t, res.Payload.Settings.SttEnabled)
+		assert.True(t, res.Payload.Settings.TelemetryEnabled)
+		assert.Empty(t, err)
+
+		resp, err := managementClient.Default.SecurityChecks.StartSecurityChecks(nil)
+		require.NoError(t, err)
+		assert.NotNil(t, resp)
+
+		client, err := api.NewClient(api.Config{
+			Address: promAddress,
+		})
+		require.NoError(t, err)
+		promClient := promapi.NewAPI(client)
+
+		testCases := []struct {
+			query string
+			len   int
+		}{
+			{
+				query: "pmm_managed_checks_alerts_generated_total",
+				len:   7,
+			},
+			{
+				query: "pmm_managed_checks_scripts_executed_total",
+				len:   3,
```

Comment:
Unfortunately, those numbers are environment-dependent. Will that work on CI?

@BupycHuk what we can do there now?

Reply:
Is this available even if we don't run STT? Looks like the count of metrics is not related to running STT checks. They will be available with 0 values. @palash25 could you check it?

Reply:
We start STT checks above (`managementClient.Default.SecurityChecks.StartSecurityChecks`) and those metrics _are_ related to STT

Reply:
>  Looks like the count of metrics is not related to running STT checks. 

Nurlan is right, I commented `resp, err := managementClient.Default.SecurityChecks.StartSecurityChecks(nil)` and the tests still pass, since we [initialize](https://github.com/percona/pmm-managed/pull/407/files#diff-e0ff53b9498062b1544f8a3a40c5eb6bR158) both the metrics with all possible labels they are initialized and available as soon as pmm-server starts. 

Should I remove the `StartSecurityChecks` call? @AlekSi

Reply:
Doh! We check result length, not value. Get it not.

I guess we can compare values too, similarly to how we do that in unit test. Once we have environment with services and real checks, we will change that test.

Reply:
cool will add the checks for values and remove the `StartSecurityChecks()` call